api/src/main/java/org/cardanofoundation/tokenmetadata/registry/api/service/Cip68FTDatumParser.java

```diff
@@ -0,0 +1,93 @@
+package org.cardanofoundation.tokenmetadata.registry.api.service;
+
+import com.bloxbean.cardano.client.plutus.spec.*;
+import com.bloxbean.cardano.client.util.HexUtil;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.cardanofoundation.tokenmetadata.registry.api.model.cip68.FungibleTokenMetadata;
+import org.springframework.stereotype.Service;
+
+import java.util.Arrays;
+import java.util.Optional;
+import java.util.stream.IntStream;
+
+@Service
+@RequiredArgsConstructor
+@Slf4j
+public class Cip68FTDatumParser {
+
+    public static final String DECIMALS = "decimals";
+    public static final String DESCRIPTION = "description";
+    public static final String LOGO = "logo";
+    public static final String NAME = "name";
+    public static final String TICKER = "ticker";
+    public static final String URL = "url";
+
+    /**
+     * Manually parses Cip68 Fungible Token Datum
+     *
+     * @param inlineDatum the hex encoded datum
+     * @return the Cip68 Fungible Token Metadata
+     */
+    public Optional<FungibleTokenMetadata> parse(String inlineDatum) {
+
+        if (inlineDatum == null || inlineDatum.trim().isEmpty()) {
```

Comment:
What scenario could this be null inline datum?

Reply:
When people put rubbish onchain. They could for instance correctly create the NFT as per standard, but "forget" to put the datum